packages/slate-editor/src/extensions/allowed-blocks/AllowedBlocksExtension.ts

```diff
@@ -0,0 +1,42 @@
+import type { Extension } from '@prezly/slate-commons';
+import { isElementNode } from '@prezly/slate-types';
+import { get } from 'lodash-es';
+import { type Editor, type NodeEntry, Transforms } from 'slate';
+
+import type { AllowedBlocksExtensionConfiguration, Block } from './types';
+
+export const EXTENSION_ID = 'AllowedBlocksExtension';
+
+export function AllowedBlocksExtension({ blocks }: AllowedBlocksExtensionConfiguration): Extension {
```

Comment:
Even though this extension's purpose right now is to normalize and enforce a certain content structure, I didn't want to call it a normalizing extension since the implemented behaviour is pretty "dumb" right now so anything more advanced will require thinking about this more.

Reply:
Hmm, this looks really nice!